test/integ/pandas/test_pandas_tools.py

```diff
@@ -219,6 +219,79 @@ def test_write_pandas(
             cnx.execute_string(drop_sql)
 
 
+@pytest.mark.parametrize("table_type", ["", "temp", "temporary", "transient"])
+def test_write_pandas_table_type(
+    conn_cnx: Callable[..., Generator[SnowflakeConnection, None, None]],
+    table_type: str,
+):
+    with conn_cnx() as cnx:
+        table_name = f"driver_versions_{table_type}"
```

Comment:
You should use the name randomizer for table names: https://github.com/snowflakedb/snowflake-connector-python/blob/0aec3eff1e4d2fe0f3845814645b7a66246a88f4/test/integ/pandas/test_pandas_tools.py#L54

```suggestion
        table_name = random_string(5, "write_pandas_table_type_")
```